content/blog/text-rendering-updates/index.Rmd

```diff
@@ -0,0 +1,171 @@
+
+While it is nice to have good access to the font files on your computer, the files has to come from somewhere. Nowadays that *somewhere* is usually [Google Fonts](https://fonts.google.com) or some other online font repository. systemfonts is now aware of a few of these repositories (Google Fonts and [Font Squirrel](https://www.fontsquirrel.com) for now), and can search and download from these (using `search_web_fonts()`, `get_from_google_fonts()`, and `get_from_font_squirrel()`). The downloaded fonts are automatically added using `add_fonts()` so they are immediately available, and by default they are placed in `~/fonts` so that they persist across R sessions and projects.
+
+But what if you don't want to think too much about all these details and just want to ensure that some specific font is available when a piece of code is running? In that case `require_font()` got you covered. This function allows you to state a dependency on a font in a script. The function scans the available fonts on the system and, if it doesn't find anything, proceeds to look for the font in the online repositories, downloading it if it finds it. If that also fails the function will either throw an error, or map the required font to a fallback of your choosing:
+
+```{r}
+#| fig.height: 3
+library(systemfonts)
+require_font("Rubik Moonrocks")
+
+plot.new()
+text(0.5, 0.5, "Fancy Font", family = "Rubik Moonrocks", cex = 6)
+```
+
+Remember that all of these niceties only goes into effect if you use a graphics device that uses systemfonts. For now that more or less means that you should use ragg (you should use ragg anyway so that is not a terrible requirement).
+
+## Getting to the Glyphs
+
+Most fonts these days are based on a vector outline. That means that they can be scaled smoothly to any size and doesn't take up a lot of storage space. It also means that there are polygons inside the font files and that these can be extracted! This is now possible with systemfonts and the new `glyph_outline()` and `glyph_raster()` functions.
+
+```{r}
+#| fig.width: 4.5
+#| fig.height: 4.5
+# Get the location of the glyph inside the font
+moonrocks <- font_info("Rubik Moonrocks")
+G <- glyph_info("G", path = moonrocks$path, index = moonrocks$index)
+
+# Extract the outline of the glyph and plot it
+outline <- glyph_outline(G$index, moonrocks$path, moonrocks$index, size = 400)
+grid::grid.path(
+  x = outline$x,
+  y = outline$y + 20, # To raise the baseline a bit
+  id = outline$contour,
+  default.units = "bigpts",
+  gp = grid::gpar(fill = "grey", col = "black", lwd = 4)
+)
+```
+
+Extracting them as polygons means that we can do all sorts of weird stuff with them if we so pleases:
```

Comment:
Once you have access to the glyph outlines, you can manipulate them programmatically. For example, here's how to create a custom italic effect[^1]:

[^1]: Real italic fonts are specifically designed for optimal appearance, rather than just applying a mathematical transformation. You should always use them if available.